analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant.RoslynCfg.cs

```diff
@@ -0,0 +1,111 @@
+            {
+                this.cfg = cfg;
+                this.memberToCheck = memberToCheck;
+            }
+
+            // Returns true if the block contains assignment before access
+            protected override bool IsValid(BasicBlock block) =>
+                Match(block, checkReadBeforeWrite: false);
+
+            // Returns true if the block contains access before assignment
+            protected override bool IsInvalid(BasicBlock block) =>
+                Match(block, checkReadBeforeWrite: true);
+
+            private bool Match(BasicBlock block, bool checkReadBeforeWrite)
+            {
+                foreach (var operation in block.OperationsAndBranchValue)
+                {
+                    if (ProcessOperation(operation, cfg, checkReadBeforeWrite, out var result))
+                    {
+                        return result;
+                    }
+                }
+
+                return false;
+            }
+
+            private bool ProcessOperation(IOperation operation, ControlFlowGraph controlFlowGraph, bool checkReadBeforeWrite, out bool result)
+            {
+                foreach (var child in new[] { operation }.ToReversedExecutionOrder())
+                {
+                    if (checkReadBeforeWrite && child.Instance.Kind == OperationKindEx.FlowAnonymousFunction)
+                    {
+                        var anonymousFunctionCfg = controlFlowGraph.GetAnonymousFunctionControlFlowGraph(IFlowAnonymousFunctionOperationWrapper.FromOperation(child.Instance));
+                        foreach (var subOperation in anonymousFunctionCfg.Blocks.SelectMany(x => x.OperationsAndBranchValue))
+                        {
+                            if (ProcessOperation(subOperation, anonymousFunctionCfg, checkReadBeforeWrite, out result))
+                            {
+                                return true;
+                            }
+                        }
+                    }
+                    else if (memberToCheck.Equals(MemberSymbol(child.Instance)))
+                    {
+                        result = IsReadOrWrite(child, checkReadBeforeWrite);
+                        return true;
+                    }
+                }
+
+                result = false;
+                return false;
+            }
+
+            private bool IsReadOrWrite(IOperationWrapperSonar child, bool checkReadBeforeWrite)
+            {
+                if (child.Instance.IsOutArgumentReference())
+                {
+                    // it is out argument - that means that this is write
+                    return !checkReadBeforeWrite;
+                }
+
+                var isWrite = child.Parent is { Kind: OperationKindEx.SimpleAssignment } parent && ISimpleAssignmentOperationWrapper.FromOperation(parent).Target == child.Instance;
+                return checkReadBeforeWrite ^ isWrite;
+            }
+
+            private static ISymbol MemberSymbol(IOperation operation) =>
```

Comment:
We will also need to check if the fields is from `this` or something else. We have a FP in this case:
```
    class Person16
    {
        int a = 42;

        public Person16(Person16 other)
        {
            other.a = 0;
        }
    }
```

Reply:
Suggested FP was addressed for three cases